packages/model-viewer/src/features/animation.ts

```diff
@@ -103,8 +103,12 @@ export const AnimationMixin = <T extends Constructor<ModelViewerElementBase>>(
     [$tick](_time: number, delta: number) {
       super[$tick](_time, delta);
 
-      if (this[$paused]) {
-        return;
+      if (this[$paused]) { 
+        // TODO we should return here if currentTime has not actually changed
+        // should we defer setting this[$scene].model.animationTime until here
+        // (store it in a temporary variable when currentTime setter is called?)
+        // return; 
+        this[$renderer].threeRenderer.shadowMap.needsUpdate = true;
```

Comment:
I think you don't need any changes here. Instead, move this line up into the `set currentTime` function. Then [here](https://github.com/google/model-viewer/blob/007ae3b9405a2dabd9a4e37b3361c7cf6729285a/packages/model-viewer/src/three-components/Model.ts#L159) you'll want to add a call to this [method](https://threejs.org/docs/index.html#api/en/animation/AnimationMixer.setTime). Let me know if that helps.

Reply:
Thanks for pointing me directly to the right places :) I think what I wasn't aware is that setting the mixer will update the animation directly without the updateAnimation call.